### PR TITLE
Change wildcards ('{*}') in GodMode to specific identifiers like ${questionid} #4402

### LINKED ISSUE #4402: Change wildcards ('{*}') in GodMode to specific identifiers like ${questionid}
Convert as many as possible so that instances where GodMode fails to behave properly are easily identified. It also leads in to #4383

### PR DESCRIPTION
**Pull request by thyageshm**
fixes #4402 
@LowWeiLin do you think these identifiers are okay? once we settle the ids, I will regenerate all the htmls

### CODE REVIEW

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -822,24 +822,24 @@ private static String processPageSourceForGodMode(String content) {
                 .replace(TestProperties.inst().TEST_UNREG_ACCOUNT, "${test.unreg}")
                 .replace(Config.SUPPORT_EMAIL, "${support.email}")
                 // today's date
-                .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "{*}")
-                .replace(TimeHelper.formatDate(now), "{*}")
+                .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "${sanitized.today}")
+                .replace(TimeHelper.formatDate(now), "${today}")
                 // now (used in comments last edited date) e.g. [Thu, 07 May 2015, 07:52:13 UTC]
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC", "{*}")
+                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC", "${comment.day.month}")
```
Comment (LowWeiLin) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: just comment.date?

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 2]
```diff
@@ -822,24 +822,24 @@ private static String processPageSourceForGodMode(String content) {
...
                 // now (used in opening time/closing time Grace period)
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}", "{*}")
+                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}", "${grace.period.date}")
                 // now (used in comments last edited date) e.g. [Thu May 07 07:52:13 UTC 2015]
-                .replaceAll(new SimpleDateFormat("EEE MMM dd ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC [0-9]{4}", "{*}")
+                .replaceAll(new SimpleDateFormat("EEE MMM dd ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC [0-9]{4}", "${comment.month.day}")
                 // dynamic feedback submission numbers
-                .replaceAll("(?s)<span class=\"submissionsNumber\".*?</span>", "<span class=\"submissionsNumber\" id=\"submissionsNumber\">{*}</span>")
+                .replaceAll("(?s)<span class=\"submissionsNumber\".*?</span>", "<span class=\"submissionsNumber\" id=\"submissionsNumber\">${submissions.made}</span>")
```
Comment (LowWeiLin) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: submissions.number

**Comment by LowWeiLin** [comment]
Some comments added. I'm okay with the it, but any specific reason why we use dots in the identifiers?

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 2 above)
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: will change this

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 1 above)
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I had it this way as there were 2 different comments dates. I think #4032 will address the issue. I will change this one

**Comment by thyageshm** [comment]
Just continued with the existing placeholder formats. They use dots to separate keywords in them like ${test.url} (although it does show hierarchy in their cases).

**Comment by wkurniawan07** [comment]
Can continue here, #4032 is already resolved.

**Comment by thyageshm** [comment]
@LowWeiLin, @wkurniawan07 I have made the changes to the files. Please look through the updates. I have tried and gone through all the html files to make sure only expected changes have occurred.

**Comment by LowWeiLin** [comment]
```
//regkey and questionid (same regex for both)
.replaceAll("value=\"([a-zA-Z0-9-_]){50,}\"","value=\"{*}\"")
```
Can change this one so it does not become {*}?
Also, the repeat range seems strange, as there is another one for question id with 62.
```
//questionid regex in responseid
.replaceAll("\"([a-zA-Z0-9-_]){62,}%", "\"\\${question\\.id}%")
```

**Comment by thyageshm** [comment]
so should I change
```
.replaceAll("value=\"([a-zA-Z0-9-_]){50,}\"","value=\"{*}\"")
```
to 
```
.replaceAll("value=\"([a-zA-Z0-9-_]){62,}\"","value=\"\\${question\\.id}\"")
.replaceAll("value=\"([a-zA-Z0-9-_]){50,}\"","value=\"\\${regkey\\.enc}\"")
```
?

**Comment by thyageshm** [comment]
Also, the second one is for questionid that is found as part of responseid

**Comment by thyageshm** [comment]
@LowWeiLin I have updated it

**Comment by LowWeiLin** [comment]
Looks good to me!

**Pull request merged**